File: src/main/java/org/techous/rainbow/Procedures.java
package org.techous.rainbow;

interface Procedures
{
    public String get(String value);

    public String getFormat(String format);
    public String getColor(String color);
    public String getBG(String background);
    public String getFormatColor(String format, String color);
    public String getFormatBG(String format, String background);
    public String getColorBG(String color, String background);
    public String getRainbow(String format, String color, String background);

    public String getFormatString(String format, String str, boolean end);
    public String getColorString(String color, String str, boolean end);
    public String getBGString(String background, String str, boolean end);
    public String getFormatColorString(String format, String color, String str, boolean end);
    public String getFormatBGString(String format, String background, String str, boolean end);
    public String getColorBGString(String color, String background, String str, boolean end);
    public String getRainbowString(String format, String color, String background, String str, boolean end);
}

File: docs/js/procedures.js
let rainbowLogo = document.getElementById('rbw-logo');
let rainbowNavScreen = document.getElementById("navScreen");
let screenFlag = false;
let rainbowNavList = document.getElementById('navlist');
let navLogo = document.getElementById('rbw-logo');
let closebtn = document.getElementById('closebtn');
let isMobile = /iPhone|iPad|iPod|Android/i.test(navigator.userAgent);

rainbowLogo.addEventListener("click", navigation);
rainbowNavList.addEventListener('click', navigation);

function navigation(){
    if(screenFlag){
        screenFlag = false;
        closeNavigation();
    }
    else
    {
        screenFlag = true;
        openNavigation();
    }
}

function openNavigation(){
    rainbowNavScreen.style.height = "100%";
    const body = document.body;
    body.style.overflowY = 'hidden';

    // manage index nav icon according to desktop browser
    if(window.innerWidth>=1200 && !isMobile){
        navLogo.style.right = "30px";
        closebtn.style.right = "30px";
    }
}

function closeNavigation(){
    rainbowNavScreen.style.height = "0%";
    const body = document.body;
    body.style.overflowY = 'scroll';

    if(window.innerWidth>=1200){
        navLogo.style.right = "15px";
    }
}

let init =()=>{
    searchArray.push("get");
    searchArray.push("getFormat");
    searchArray.push("getColor");
    searchArray.push("getBG");
    searchArray.push("getFormatColor");
    searchArray.push("getFormatBG");
    searchArray.push("getColorBG");
    searchArray.push("getRainbow");
    searchArray.push("getFormatString");
    searchArray.push("getColorString");
    searchArray.push("getBGString");
    searchArray.push("getFormatColorString");
    searchArray.push("getFormatBGString");
    searchArray.push("getColorBGString");
    searchArray.push("getRainbowString");
}